proto/tendermint/types/types.proto

```diff
@@ -79,6 +79,8 @@ message Header {
   // consensus info
   bytes evidence_hash    = 13;  // evidence included in the block
   bytes proposer_address = 14;  // original proposer of the block
+
+  bytes app_data = 15; // arbitrary bytes set by the aplication, nondeterministic
```

Comment:
Few things I want to bring up about this change: 
1) Has this amendment been added to the spec? 
2) What is the justification of including arbitrary application defined bytes in the header? Why can't we just use the app hash to append any extra data the application wants (I don't think there are any size limits on the app hash)
3) What is meant here by non-deterministic? 

I know that having such a field allows for privileged application data to be directly verified via the light client. I'm just uncertain about having application specific data in the tendermint header. I think I'm mostly for such a transition but I would like to gather greater consensus first. Perhaps this is something we can bring up in the tendermint dev call